client/src/components/MultipleChooserPanel/index.js

```diff
@@ -30,6 +51,12 @@ export class MultipleChooserPanel extends InlinePanel {
         },
         { multiple: true },
       );
+      if (opts.allowDuplicates === 'True') {
+        openModalButton.setAttribute(
```

Comment:
All information that needs to be passed from the calling page to the chooser modal should be passed in the URL, not attached to a button for the modal to pick up later.

As I said before, it's really important in a project the size of Wagtail that we build things out of independent components with clear inputs and outputs, that don't have to know the details of how other components behave. For the chooser modal, the URL should tell it everything it needs to know about how it should behave - it shouldn't have to know about the fact that it was triggered by a user clicking a button, and that button having extra information attached to it.

This also makes it easier to write test code. If the chosen IDs are passed as part of the URL, we can write a test that calls the chooser at the appropriate URL and verifies that the relevant checkboxes are disabled.

Now, since the URL handling and the rendering of the checkboxes both happen in Python code, this probably means that disabling the checkboxes will be done in Python code rather than Javascript. I think this is the correct approach - the disabled state of the checkboxes is part of the initial state of the page, and not something that changes in response to user interaction, so it makes sense for it to be done in Python code.

Reply:
Good point 

> since the URL handling and the rendering of the checkboxes both happen in Python code, this probably means that disabling the checkboxes will be done in Python code rather than Javascript

I moved it to the server side accordingly. and Thanks for the detailed explanation as well.